pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -66,44 +65,27 @@ func (s *preFilterState) Clone() framework.StateData {
 	copy.topologyToMatchedAffinityTerms = s.topologyToMatchedAffinityTerms.clone()
 	copy.topologyToMatchedAntiAffinityTerms = s.topologyToMatchedAntiAffinityTerms.clone()
 	copy.topologyToMatchedExistingAntiAffinityTerms = s.topologyToMatchedExistingAntiAffinityTerms.clone()
+	// No need to deep copy the podInfo because it shouldn't change.
+	copy.podInfo = s.podInfo
 
 	return &copy
 }
 
 // updateWithPod updates the preFilterState counters with the (anti)affinity matches for the given pod.
-func (s *preFilterState) updateWithPod(updatedPod, pod *v1.Pod, node *v1.Node, multiplier int64) error {
+func (s *preFilterState) updateWithPod(updatedPod *v1.Pod, node *v1.Node, multiplier int64) error {
 	if s == nil {
 		return nil
 	}
 
 	// Update matching existing anti-affinity terms.
-	updatedPodAffinity := updatedPod.Spec.Affinity
-	if updatedPodAffinity != nil && updatedPodAffinity.PodAntiAffinity != nil {
-		antiAffinityTerms, err := getAffinityTerms(pod, schedutil.GetPodAntiAffinityTerms(updatedPodAffinity.PodAntiAffinity))
-		if err != nil {
-			return fmt.Errorf("error in getting anti-affinity terms of Pod %v: %v", updatedPod.Name, err)
-		}
-		s.topologyToMatchedExistingAntiAffinityTerms.updateWithAntiAffinityTerms(pod, node, antiAffinityTerms, multiplier)
-	}
+	// TODO(#91058): AddPod/RemovePod should pass a *framework.PodInfo type instead of *v1.Pod.
+	updatedPodInfo := framework.NewPodInfo(updatedPod)
+	s.topologyToMatchedExistingAntiAffinityTerms.updateWithAntiAffinityTerms(s.podInfo.Pod, node, updatedPodInfo.RequiredAntiAffinityTerms, multiplier)
 
 	// Update matching incoming pod (anti)affinity terms.
-	affinity := pod.Spec.Affinity
-	podNodeName := updatedPod.Spec.NodeName
-	if affinity != nil && len(podNodeName) > 0 {
-		if affinity.PodAffinity != nil {
-			affinityTerms, err := getAffinityTerms(pod, schedutil.GetPodAffinityTerms(affinity.PodAffinity))
-			if err != nil {
-				return fmt.Errorf("error in getting affinity terms of Pod %v: %v", pod.Name, err)
-			}
-			s.topologyToMatchedAffinityTerms.updateWithAffinityTerms(updatedPod, node, affinityTerms, multiplier)
-		}
-		if affinity.PodAntiAffinity != nil {
-			antiAffinityTerms, err := getAffinityTerms(pod, schedutil.GetPodAntiAffinityTerms(affinity.PodAntiAffinity))
-			if err != nil {
-				klog.Errorf("error in getting anti-affinity terms of Pod %v: %v", pod.Name, err)
-			}
-			s.topologyToMatchedAntiAffinityTerms.updateWithAntiAffinityTerms(updatedPod, node, antiAffinityTerms, multiplier)
-		}
+	if len(updatedPod.Spec.NodeName) > 0 {
```

Comment:
I feel like it's not needed as `updatePod` is from snapshot which literally means assignedPod. Otherwise, L83 should be also guarded by `if len(updatedPod.Spec.NodeName) > 0`.

Reply:
sg, done.